### PR TITLE
miscellaneous fixes to azure cli. added build folders to gitignore.  added a readme

### PR DESCRIPTION
**Pull request by swagatmishra2007**
bugs fixed-
a) made description optional while creating bots
b) az download now shows the path to which it was downloaded
c) better error handling
d) added a readme.md file to the project.

### CODE REVIEW

File: AzureCli/readme.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+## Creating a Bot
+Login into your azure account via cli :   
+`az login`  
+Ensure that you are using the right subscription for creating your bot -   
+`az account show`  
+Change your subscription , if needed  
+`az account set --subscription <Your-subscription-name>`  
+
+You need an existing resource group to create your bot and related assets. Either use an existing one or create one.
```
Comment (vishwacsena) [inline AzureCli/readme.md]: You need a resource group (existing or new) to create your bot and related assets.

File: AzureCli/readme.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+on the CLI you can create a new resource group as -  
+`az group create -l <resource-group-location> -n <resource-group-name>`
+
+Now create your bot -  
+`az bot create -k webapp -g myResourceGroup -n myBot`  
+If you did not specify a microsoft app credentials during creation, you will be asked to login again so that we can create those credentials on your behalf.
```
Comment (vishwacsena) [inline AzureCli/readme.md]: Tip:
You can set defaults for resource group using az configure --defaults group=<your group name>

File: AzureCli/readme.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+you can access your bot credentials and information using   
+`az bot show --msbot -n myBot -g myResourceGroup`
```
Comment (vishwacsena) [inline AzureCli/readme.md]: You can get your bot credentials and related information using the --msbot option.

File: AzureCli/readme.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+1) To avoid having to type your bot name and resource group name everytime, use
+    `az configure --defaults botname=myBot group=myResourceGroup`
+2) If you are stuck at any point, add a `-h` to the command to help you out  
+Eg - `az bot create -h`  
+
+## Modifying a Bot
+The following workflow assumes that you have your bot name and resource group name configured by default using 
+az configure.  
+A typical bot workflow would be connecting your bot to msbot, downloading the source code and then publishing back to azure with changes.
+First connect your bot to msbot
```
Comment (vishwacsena) [inline AzureCli/readme.md]: You can get the MSBOT CLI from [here](https://github.com/Microsoft/botbuilder-tools/tree/master/MSBot)

File: AzureCli/readme.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+`az account show`  
+Change your subscription , if needed  
+`az account set --subscription <Your-subscription-name>`  
+
+You need an existing resource group to create your bot and related assets. Either use an existing one or create one.
+on the CLI you can create a new resource group as -  
+`az group create -l <resource-group-location> -n <resource-group-name>`
+
+Now create your bot -  
+`az bot create -k webapp -g myResourceGroup -n myBot`
```
Comment (vishwacsena) [inline AzureCli/readme.md]: When I tried this yesterday, --description was required. Did we fix that? If not, we should include description here.

File: AzureCli/readme.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+A typical bot workflow would be connecting your bot to msbot, downloading the source code and then publishing back to azure with changes.
+First connect your bot to msbot  
+`az bot show --msbot | msbot connect Azure --stdin`  
+This will update your .bot file with all the information needed by other botbuilder tools to access it.  
+Now download the source code  
+`az bot download`  
+This creates a folder with your bot name in the current directory (use `--file-save-path` to override), and downloads the bot source code into this folder. If you created your bot using `az bot create`, then the source code will be an echo bot project. It contains postdeploy scripts and build scripts to help deploy it back to azure. Modify this template code as per your needs.  
+To publish your code back to azure, from within the bot directory(use `--code-dir` to override) run   
+`az bot publish`
+
```
Comment (vishwacsena) [inline AzureCli/readme.md]: Add a section here with different publish options include GitHub integration

File: AzureCli/readme.md (hunk 5 above)
Comment (swagatmishra2007) [reply AzureCli/readme.md]: description should be optional. Maybe the package was not udpated when you trried. let me try again and verify

File: AzureCli/readme.md (hunk 6 above)
Comment (swagatmishra2007) [reply AzureCli/readme.md]: Added a section for github integration. Direct source code publish is already covered in this section.

File: AzureCli/azext_bot/azext_bot/custom.py [hunk 7]
```diff
@@ -227,7 +233,7 @@ def publish_app(cmd, client, resource_group_name, resource_name, giturl = None,
     if code_dir:
         import os
         if not os.path.isdir(code_dir):
-            raise CLIError('plesae supply a valid directory path containing your source code')
+            raise CLIError('please supply a valid directory path containing your source code')
```
Comment (carlosscastro) [inline AzureCli/azext_bot/azext_bot/custom.py]: Upper case: Please supply a..

File: AzureCli/azext_bot/azext_bot/custom.py [hunk 8]
```diff
@@ -238,10 +244,22 @@ def publish_app(cmd, client, resource_group_name, resource_name, giturl = None,
...
+    )
+    if(raw_bot_properties.properties.kind == 'bot'):
+        raise CLIError('source download is not supported for registration only bots')
     import os
     file_save_path = file_save_path or os.getcwd()
     if not os.path.isdir(file_save_path):
         raise CLIError('path name not valid')
+    folder_path = os.path.join(file_save_path,resource_name)
+    if os.path.exists(folder_path):
+        raise CLIError('the path {0} already exists. Please delete it or specify an alternate path'.format(folder_path))
```
Comment (carlosscastro) [inline AzureCli/azext_bot/azext_bot/custom.py]: Upper case: The path {0}...

This applies to all user-visible strings
